controller/src/main/java/org/jboss/as/controller/security/CredentialReference.java

```diff
@@ -0,0 +1,417 @@
+     */
+    public static void readCredentialReference(ModelNode addOperation, XMLExtendedStreamReader reader) throws XMLStreamException {
+        AttributeParser.OBJECT_PARSER.parseElement(CredentialReference.getAttributeDefinition(), reader, addOperation);
+    }
+
+    static class ClearTextCredentialStore extends CredentialStore {
+        private static String TYPE = "ClearTextCredentialStore";
+        private CredentialReference credentialReference;
+
+        ClearTextCredentialStore(CredentialReference credentialReference) {
+            super(null, null, TYPE);
+            this.credentialReference = credentialReference;
+        }
+
+        /**
+         * Checks whether underlying credential store is initialized.
+         *
+         * @return {@code true} in case of initialization passed successfully, {@code false} otherwise.
+         */
+        @Override
+        public boolean isInitialized() {
+            return true;
+        }
+
+        /**
+         * Check if credential store supports modification of actual store
+         *
+         * @return true in case of modification of store is supported
+         */
+        @Override
+        public boolean isModifiable() {
+            return false;
+        }
+
+        /**
+         * Check whether credential store has an entry associated with the given credential alias of specified credential type.
+         *
+         * @param credentialAlias alias to check existence
+         * @param credentialType  to check existence in the credential store
+         * @return true in case key exist in store
+         * @throws CredentialStoreException           when there is a problem with credential store
+         * @throws UnsupportedCredentialTypeException when the credentialType is not supported
+         */
+        @Override
+        public <C extends Credential> boolean exists(String credentialAlias, Class<C> credentialType) throws CredentialStoreException, UnsupportedCredentialTypeException {
+            return false;
+        }
+
+        /**
+         * Store credential to the store under the given alias. If given alias already contains specific credential type type the credential
+         * replaces older one. <em>Note:</em> {@link CredentialStoreSpi} supports storing of multiple entries (credential types) per alias.
+         * Each must be of different credential type.
+         *
+         * @param credentialAlias to store the credential to the store
+         * @param credential      instance of {@link Credential} to store
+         * @throws CredentialStoreException           when the credential cannot be stored
+         * @throws UnsupportedCredentialTypeException when the credentialType is not supported
+         */
+        @Override
+        public <C extends Credential> void store(String credentialAlias, C credential) throws CredentialStoreException, UnsupportedCredentialTypeException {
+            throw new RuntimeException("method not implemented");
```

Comment:
Should this be one of the 2 documented exception types?

I only care if this class isn't private.